test/e2e/apps/job.go

```diff
@@ -361,6 +368,172 @@ var _ = SIGDescribe("Job", func() {
+					job.ObjectMeta.Namespace == ns &&
+					job.Annotations["updated"] == "true" &&
+					event.Type == watch.Modified
+				if !found {
+					framework.Logf("Event %v observed for Job %v in namespace %v with annotations: %v", event.Type, job.ObjectMeta.Name, job.ObjectMeta.Namespace, job.Annotations)
+					return false, nil
+				}
+				framework.Logf("Event %v found for Job %v in namespace %v with annotations: %v", event.Type, job.ObjectMeta.Name, job.ObjectMeta.Namespace, job.Annotations)
+				return found, nil
+			}
+			framework.Logf("Observed event: %+v", event.Object)
+			return false, nil
+		})
+		if err != nil {
+			p, _ := jobClient.Get(context.TODO(), jobName, metav1.GetOptions{})
+			framework.Logf("Job: %+v", p)
+		}
+		framework.ExpectNoError(err, "failed to see Job %v in namespace %v running", updatedJob.ObjectMeta.Name, ns)
+
+		ginkgo.By("Listing all Jobs with LabelSelector")
+		jobs, err := f.ClientSet.BatchV1().Jobs("").List(context.TODO(), metav1.ListOptions{LabelSelector: labelSelector})
+		framework.ExpectNoError(err, "Failed to list job. %v", err)
+		framework.ExpectEqual(len(jobs.Items), 1, "Failed to find job %v", jobName)
+		testJob := jobs.Items[0]
+		framework.Logf("Job: %v as labels: %v", testJob.Name, testJob.Labels)
+
+		ginkgo.By("Waiting for job to complete")
+		err = e2ejob.WaitForJobComplete(f.ClientSet, ns, jobName, completions)
+		framework.ExpectNoError(err, "failed to ensure job completion in namespace: %s", ns)
+
+		ginkgo.By("Delete a job collection with a labelselector")
+		propagationPolicy := metav1.DeletePropagationBackground
+		err = f.ClientSet.BatchV1().Jobs(ns).DeleteCollection(context.TODO(), metav1.DeleteOptions{PropagationPolicy: &propagationPolicy}, metav1.ListOptions{LabelSelector: labelSelector})
+		framework.ExpectNoError(err, "failed to delete job %s in namespace: %s", job.Name, ns)
+
+		ginkgo.By("Watching for Job to be deleted")
+		ctx, cancel = context.WithTimeout(context.Background(), f.Timeouts.PodStart)
+		defer cancel()
+		_, err = watchtools.Until(ctx, jobsList.ResourceVersion, w, func(event watch.Event) (bool, error) {
+			if job, ok := event.Object.(*batchv1.Job); ok {
+				found := job.ObjectMeta.Name == testJob.ObjectMeta.Name &&
+					job.ObjectMeta.Namespace == ns &&
+					job.Labels["e2e-job-label"] == jobName &&
+					event.Type == watch.Deleted
+				if !found {
+					framework.Logf("Event %v observed for Job %v in namespace %v with labels: %v", event.Type, job.ObjectMeta.Name, job.ObjectMeta.Namespace, job.Labels)
+					return false, nil
+				}
+				framework.Logf("Event %v found for Job %v in namespace %v with labels: %v", event.Type, job.ObjectMeta.Name, job.ObjectMeta.Namespace, job.Labels)
+				return found, nil
+			}
+			framework.Logf("Observed event: %+v", event.Object)
+			return false, nil
+		})
+		if err != nil {
+			p, _ := jobClient.Get(context.TODO(), jobName, metav1.GetOptions{})
+			framework.Logf("Job: %+v", p)
```

Comment:
What's the intent of this log line?

Reply:
Provide more detail about the Job if the required event was not found. There have been a number of issues trying to confirm what has caused a test to flake when dealing with watch events.

Reply:
Can you include some context in the message? Perhaps:

```
framework.Logf("Job still exists after deletion call: %+v", p)
```

Also, why is the variable named `p`? Should it be `j` for job?

Reply:
Slipped though from one of my other status tests. Fixed variable and message for each watch scenario.